src/index.js

```diff
@@ -159,13 +159,27 @@ export function app(props, container) {
     }
   }
 
-  function removeElement(parent, element, props) {
+  function destroyNode(element, node) {
+    if (node.children) {
+      for (var i = 0; i < node.children.length; i++) {
+        if (typeof node.children[i] !== "string") {
+          destroyNode(element.childNodes[i], node.children[i])
+        }
+      }
+    }
+    if (node.props && node.props.ondestroy) {
+      node.props.ondestroy(element)
+    }
+  }
+
+  function removeElement(parent, element, node) {
     function done() {
       parent.removeChild(element)
+      destroyNode(element, node)
```

Comment:
Could we call `destroyNode` before the element is removed instead?

I'm not sure, but I suspect the child elements may be in a somehow less usable state when their tree is detached from the DOM, potentially causing issues for the ondestroy handlers.

Reply:
What issues do you imagine those would be?

Reply:
I can't say off-hand -- but I know I've run into trouble before with elements not being able to do everything I expect, when they're not mounted in the dom. But maybe those things wouldn't be relevant here. 

Anyhow, is there a reason for the ordering or is it arbitrary? If it's arbitrary I'd feel safer putting destroyNode first. That's all

Reply:
> What issues do you imagine those would be?

For example unable to save last element dimensions because offsetHeight will return 0 for detached node.